packages/relay/src/lib/precheck.ts

```diff
@@ -186,26 +186,24 @@ export class Precheck {
   }
 
   /**
-   * Calculates the intrinsic gas cost based on the number of empty bytes and whether the transaction is CONTRACT_CREATE
+   * Calculates the intrinsic gas cost based on the number of bytes in the data field
    * @param data
-   * @param to
    * @private
    */
-  private static transactionIntrinsicGasCost(data: string, to: string | undefined) {
-    const isCreate = (to == undefined) || (to.length == 0);
-
+  private static transactionIntrinsicGasCost(data: string) {
+    let trimmedData = data.replace("0x",'');
+  
     let zeros = 0;
-
-    const dataBytes = Buffer.from(data, 'hex');
-
-    for (const c of dataBytes) {
-      if (c == 0) {
-        zeros++;
-      }
+    let nonZeros = 0;
+    for (let index = 0; index < trimmedData.length; index += 2) {
+        const bytes = trimmedData[index] + trimmedData[index + 1];
+        if (bytes === "00") {
+            zeros++;
+        } else {
+            nonZeros++;
+        }
     }
 
-    const nonZeros = data.replace('0x', '').length - zeros;
-    const cost = constants.TX_BASE_COST + constants.TX_DATA_ZERO_COST * zeros + constants.ISTANBUL_TX_DATA_NON_ZERO_COST * nonZeros;
-    return isCreate ? cost + constants.TX_CREATE_EXTRA : cost;
+    return constants.TX_BASE_COST + constants.TX_DATA_ZERO_COST * zeros + constants.ISTANBUL_TX_DATA_NON_ZERO_COST * nonZeros;
```

Comment:
One question, what is the `IntrinsicGasCost`? and why is it calculated with this formula? 
I thought that accurate GasCost was possible only using contracts/call?

Reply:
This is like the minimum amiunt of gas needed to execute the transaction, based on the data provided and base cost for every transaction. The real and more accurate gas estimation is done by the web3 module. Here we are just making sure that the provided gasLimit is covering the bare minimum for this tx. Calucalation is done based on the info from Hedera docs, but basicaly for every empty byte we multiply by 4, and for every non empty we multiply by 16 and add all this to the base cost (21000).

Reply:
Thanks for the explanation, it makes sense 👍